src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -608,81 +608,85 @@ bool GetPeerNodeInfo(CommandHandler * commandHandler, const ConcreteCommandPath
-                    MutableCharSpan uri(uriBuffer);
-                    err = bdx::MakeURI(targetNodeId.GetNodeId(), AsCharSpan(data.imageURI), uri);
-                    if (CHIP_NO_ERROR != err) {
-                        LogErrorOnFailure(err);
-                        handler->AddStatus(cachedCommandPath, Protocols::InteractionModel::Status::Failure);
-                        handle.Release();
-                        gOtaSender.ResetState();
-                        return;
-                    }
+                      CommandHandler * handler = EnsureValidState(handle, cachedCommandPath, "QueryImage", data, error);
+                      VerifyOrReturn(handler != nullptr);
 
-                    response.imageURI.SetValue(uri);
-                    handler->AddResponse(cachedCommandPath, response);
-                    handle.Release();
-                    return;
-                }
-                if (!hasUpdate) {
-                    // Send whatever error response our delegate decided on.
-                    handler->AddResponse(cachedCommandPath, response);
-                } else {
-                    // We must have isBDXProtocolSupported false.  Send the corresponding error status.
-                    Commands::QueryImageResponse::Type protocolNotSupportedResponse;
-                    protocolNotSupportedResponse.status
-                        = static_cast<OTAQueryStatus>(MTROTASoftwareUpdateProviderOTAQueryStatusDownloadProtocolNotSupported);
-                    handler->AddResponse(cachedCommandPath, protocolNotSupportedResponse);
-                }
-                handle.Release();
-                gOtaSender.ResetState();
-            }
+                      ChipLogDetail(Controller, "QueryImage: application responded with: %s",
+                          [[data description] cStringUsingEncoding:NSUTF8StringEncoding]);
 
-                          errorHandler:^(NSError *) {
-                              // Not much we can do here
-                          }];
-    };
+                      Commands::QueryImageResponse::Type delegateResponse;
+                      ConvertFromQueryImageResponseParms(data, delegateResponse);
+
+                      // If BDX protocol is not supported, send a response with status - Protocol Not supported and reset state.
+                      if (![commandParams.protocolsSupported
+                              containsObject:@(MTROtaSoftwareUpdateProviderOTADownloadProtocolBDXSynchronous)]) {
+                          Commands::QueryImageResponse::Type protocolNotSupportedresponse;
+                          protocolNotSupportedresponse.status
+                              = static_cast<OTAQueryStatus>(MTROTASoftwareUpdateProviderOTAQueryStatusDownloadProtocolNotSupported);
+                          handler->AddResponse(cachedCommandPath, protocolNotSupportedresponse);
+                          handle.Release();
+                          gOtaSender.ResetState();
```

Comment:
Two questions:

1. Why do we need a `ResetState` here?  We haven't done a `PrepareForTransfer` for this request, so the state has nothing to do with us, right?
2. Why wouldn't we just return after sending the response,  It makes it much easier to tell that nothing _after_ this long `if/else` thing will run in this case.
3. If we do return here, the `else if` following should just become `if`.

Reply:
1. ResetState is not needed here.
2. Looks cleaner with if/elseif/else to me :). Process each case and implicit return. but i agree that if someone adds some code to this block they could make mistakes if we don't return explicitly. Either is fine. Will change and keep this is mind for other PRs as well.

Reply:
not needed. cleaned up as requested